var/spack/repos/builtin/packages/py-throttler/package.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyThrottler(PythonPackage):
+    """Zero dependency Python package for easy throttling with asyncio support."""
+
+    homepage = "https://github.com/uburuntu/throttler"
+    pypi = "throttler/throttler-1.2.1.tar.gz"
+
+    maintainers = ["charmoniumQ"]
+
+    version("1.2.1", sha256="8b23d3485a96d98484024a850c1887ccc685bead17e86c8a9e4b0335e7d74778")
+
+    depends_on("py-setuptools", type="build")
+
+    patch("setup.patch")
```

Comment:
Why is this patch necessary?

Reply:
The `setup.py` references files that are in the Git repository but not the PyPI archived tarball, so the vendored `setup.py` will fail.

Reply:
Ah, I was looking at 1.2.2 (released yesterday) which fixes this. Do you want to just add 1.2.2 and drop 1.2.1 and the patch?

Reply:
Sounds good